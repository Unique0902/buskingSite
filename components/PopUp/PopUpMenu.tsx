import React, { createContext, ReactNode, useContext, useState } from 'react';
import PopupWrapper from './PopupWrapper';

interface Props {
  children: ReactNode;
}

type PopUpMenuContextProps = {
  isShowMenu: boolean;
  setIsShowMenu: React.Dispatch<React.SetStateAction<boolean>>;
};

const PopUpMenuContext = createContext<PopUpMenuContextProps>(
  {} as PopUpMenuContextProps
);

const PopUpMenu = ({ children }: Props) => {
  const [isShowMenu, setIsShowMenu] = useState<boolean>(false);
  return (
    <PopUpMenuContext.Provider value={{ isShowMenu, setIsShowMenu }}>
      {children}
    </PopUpMenuContext.Provider>
  );
};

const OuterBtn = ({ children }: { children: ReactNode }) => {
  const { setIsShowMenu } = useContext(PopUpMenuContext);
  return <button onClick={() => setIsShowMenu(true)}>{children}</button>;
};

const Inner = ({
  children,
  isPopUpInnerLeft,
}: {
  children: ReactNode;
  isPopUpInnerLeft: boolean;
}) => {
  const { isShowMenu, setIsShowMenu } = useContext(PopUpMenuContext);
  if (isShowMenu) {
    return (
      <PopupWrapper
        handleClickOther={() => {
          setIsShowMenu(false);
        }}
        isLeft={isPopUpInnerLeft}
      >
        {children}
      </PopupWrapper>
    );
  }
  return <></>;
};
PopUpMenu.OuterBtn = OuterBtn;
PopUpMenu.Inner = Inner;

export default PopUpMenu;
