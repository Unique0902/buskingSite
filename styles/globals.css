@tailwind base;
@tailwind components;
@tailwind utilities;
/* TODO: 글꼴 불러오는 로딩 줄이거나 없애기 */
* {
  box-sizing: border-box;
  font-family: 'Pretendard';
}

@font-face {
  font-family: 'Pretendard';
  font-weight: 45 920;
  font-style: normal;
  font-display: swap;
  src: url('../assets/fonts/PretendardVariable.woff2')
    format('woff2-variations');
}

li {
  list-style: none;
}
body {
  margin: 0;
  @apply text-black dark:text-white dark:bg-slate-600;
}
p,
h1,
h2 {
  margin: 0;
}
ul {
  padding: 0;
}
