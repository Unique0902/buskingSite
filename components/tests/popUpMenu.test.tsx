import { describe, expect, it } from '@jest/globals';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import PopUpMenu from '../PopUp/PopUpMenu';
import userEvent from '@testing-library/user-event';

describe('popUpMenu component test', () => {
  it('render correctly', () => {
    const component = renderer.create(
      <PopUpMenu>
        <PopUpMenu.OuterBtn>
          <div>outer btn</div>
        </PopUpMenu.OuterBtn>
        <PopUpMenu.Inner isPopUpInnerLeft>
          <div>inner</div>
        </PopUpMenu.Inner>
      </PopUpMenu>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('show inner when outerBtn is clicked', async () => {
    render(
      <PopUpMenu>
        <PopUpMenu.OuterBtn>
          <div>outer btn</div>
        </PopUpMenu.OuterBtn>
        <PopUpMenu.Inner isPopUpInnerLeft>
          <div>inner</div>
        </PopUpMenu.Inner>
      </PopUpMenu>
    );
    expect(screen.queryByText('inner')).toBeNull();
    const outerBtn = screen.getByText('outer btn');
    await userEvent.click(outerBtn);
    expect(screen.queryByText('inner')).not.toBeNull();
  });
});
