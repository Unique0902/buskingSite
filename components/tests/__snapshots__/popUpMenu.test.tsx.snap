// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`popUpMenu component test render correctly 1`] = `
<button
  onClick={[Function]}
>
  <div>
    outer btn
  </div>
</button>
`;
